src/pages/Location/Location.js

```diff
@@ -94,6 +95,11 @@ const LocationComponent = () => {
   // update regionShorthandRef
   useEffect(() => {
     regionShorthandRef.current = regionShorthand;
+    if (regionShorthand) {
+      dispatchRedux(updateRegion({
+        regionShorthand,
+      }));
+    }
```

Comment:
Redux is updated when the component will unmount so it don't need to update here. Also it would replace the redux value so if user have a custom zone in the history it would be replaced.

Reply:
this is updated here so the next button has the info it needs to be disabled or not.